src/Thinktecture.Relay.Connector.Protocols.SignalR/DiscoveryDocumentRetryPolicy.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using Microsoft.AspNetCore.SignalR.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Thinktecture.Relay.Connector.Options;
+
+namespace Thinktecture.Relay.Connector.Protocols.SignalR
+{
+	/// <inheritdoc />
+	public class DiscoveryDocumentRetryPolicy : IRetryPolicy
+	{
+		private readonly Random _random = new Random();
+		private readonly ILogger<DiscoveryDocumentRetryPolicy> _logger;
+		private readonly int _minimumDelay;
+		private readonly int _maximumDelay;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DiscoveryDocumentRetryPolicy"/> class.
+		/// </summary>
+		/// <param name="logger">An <see cref="ILogger{TCategoryName}"/>.</param>
+		/// <param name="relayConnectorOptions">An <see cref="IOptions{TOptions}"/>.</param>
+		public DiscoveryDocumentRetryPolicy(ILogger<DiscoveryDocumentRetryPolicy> logger,
+			IOptions<RelayConnectorOptions> relayConnectorOptions)
+		{
+			_logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+			if (relayConnectorOptions == null) throw new ArgumentNullException(nameof(relayConnectorOptions));
+
+			_minimumDelay = (int)relayConnectorOptions.Value.DiscoveryDocument.ReconnectMinimumDelay.TotalSeconds;
+			_maximumDelay = (int)(relayConnectorOptions.Value.DiscoveryDocument.ReconnectMaximumDelay.TotalSeconds + 1);
```

Comment:
Why +1 second?

Reply:
It's an exclusive upper bound.

Reply:
I added a comment to explain this for future "questioning".

Reply:
Ahh. Makes sense ;)